### ISSUE TITLE
`perlPackages.XMLLibXSLT` fails on M1 Mac

### ISSUE THREAD

**Issue by xave** [NixOS/nixpkgs#162390]
**Describe the bug**

Running the command  ```nix-env -iA nixpkgs.texlive.combined.scheme-full``` fails with errors on M1 Mac. 

See log.

**Steps To Reproduce**

1. Run nix-env -iA nixpkgs.texlive.combined.scheme-full
2. Failure.

**Expected Behavior**
A successful installation of the package.

**nix-env --version output**
```nix-env (Nix) 2.6.1```

**Additional context**
[texlive-nix-log.txt](https://github.com/NixOS/nix/files/8155234/texlive-nix-log.txt)

**Comment by hmenke** [comment]
You didn't even get to building TeX Live. It's perl5.34.0-XML-LibXSLT-1.99 that failed building.

**Comment by xave** [comment]
Good catch.

**Comment by janfrode** [comment]
Hitting same issue. Here's the log from the perlPackages.XMLLibXSLT build failure:

[nix-log-perl5.34.0-XML-LibXML-2.0206.txt](https://github.com/NixOS/nixpkgs/files/8348820/nix-log-perl5.34.0-XML-LibXML-2.0206.txt)

**Comment by janfrode** [comment]
Seems like a trivial failure (can't find -lz), but unfortunately I have no idea how to fix nix builds..

**Comment by janfrode** [comment]
As a workaround, I switched to nixpkgs-unstable, and there I can build the full texlive.combined.scheme-full:

**Comment by janfrode** [comment]
As a workaround, I switched to nixpkgs-unstable, and there I can build the full texlive.combined.scheme-full:

```
nix-channel --add https://nixos.org/channels/nixpkgs-unstable
nix-channel --update
nix-shell -p texlive.combined.scheme-full
```

**Comment by wegank** [comment]
Seems to be fixed on both 22.05 and unstable:

`nix shell github:NixOS/nixpkgs/nixpkgs-22.05-darwin#perl534Packages.XMLLibXSLT`
